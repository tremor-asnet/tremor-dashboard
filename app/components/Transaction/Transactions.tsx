// Icons
import { FaCalendarAlt } from "react-icons/fa";

// Components
import TransactionList from "@/components/Transaction/TransactionList/TransactionList";

// Types
import { Transaction } from "@/types";

interface TransactionsProps {
  newest: Transaction[];
  yesterday: Transaction[];
  date: string;
}

const Transactions = ({ newest, yesterday, date }: TransactionsProps) => {
  return (
    <div className="p-6 bg-white dark:bg-[#202940] rounded-lg shadow-md">
      <div className="flex justify-between">
        <h6 className="font-semibold text-[#344767] dark:text-white">
          Your Transactions
        </h6>
        <div className="flex items-center text-[#7b809a]">
          <FaCalendarAlt />
          <p className="text-sm ml-2">{date}</p>
        </div>
      </div>
      <h6 className="transaction-list-heading">NEWEST</h6>
      <TransactionList transactions={newest} />
      <h6 className="transaction-list-heading">YESTERDAY</h6>
      <TransactionList transactions={yesterday} />
    </div>
  );
};

export default Transactions;
