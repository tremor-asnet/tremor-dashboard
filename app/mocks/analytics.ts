export const SALES_BY_COUNTRY = [
  {
    id: "us",
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: "2500",
    value: "$230,900",
    bounce: "29.9%",
  },
  {
    id: "ge",
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: "3.900",
    value: "$440,000",
    bounce: "40.22%",
  },
  {
    id: "gb",
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: "1.400",
    value: "$190,700",
    bounce: "23.44%",
  },
  {
    id: "br",
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: "562",
    value: "$143,960",
    bounce: "32.14%",
  },
  {
    id: "au",
    flag: "/assets/images/flags/AU.webp",
    country: "Australia",
    sales: "400",
    value: "$143,960",
    bounce: "56.83%",
  },
];

export const STATISTICAL_DATA = [
  {
    id: "1",
    type: "Bookings",
    amount: 281,
    totalAmount: 55,
    duration: "than last week",
  },
  {
    id: "2",
    type: "Today's Users",
    amount: 2300,
    totalAmount: 3,
    duration: "than last month",
  },
  {
    id: "3",
    type: "Revenue",
    amount: 34,
    totalAmount: 1,
    duration: "than yesterday",
  },
  {
    id: "4",
    type: "Followers",
    amount: 91,
    totalAmount: 0,
    duration: "Just updated",
  },
];

export const ANALYTIC_INFO = [
  {
    id: "1",
    photo: "/assets/images/analytics/analytics-info-1.webp",
    name: "Cozy 5 Stars Apartment",
    description: `The place is close to Barceloneta Beach and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Barcelona.`,
    price: 899,
    location: "Barcelona, Spain",
  },
  {
    id: "2",
    photo: "/assets/images/analytics/analytics-info-2.webp",
    name: "Office Studio",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the night life in London, UK.`,
    price: 1.119,
    location: "London, UK",
  },
  {
    id: "3",
    photo: "/assets/images/analytics/analytics-info-3.webp",
    name: "Beautiful Castle",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Milan.`,
    price: 459,
    location: "Milan, Italy",
  },
];

export const WEBSITE_CHART = {
  id: "view-campaign",
  display: "Website Views",
  desc: "Last Campaign Performance",
  modified: "campaign sent 2 days ago",
  data: [
    {
      description: "M",
      sales: 50,
    },
    {
      description: "T",
      sales: 20,
    },
    {
      description: "W",
      sales: 10,
    },
    {
      description: "T",
      sales: 22,
    },
    {
      description: "F",
      sales: 50,
    },
    {
      description: "S",
      sales: 10,
    },
    {
      description: "S",
      sales: 40,
    },
  ],
};
