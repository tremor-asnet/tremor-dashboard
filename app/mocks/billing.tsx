// Constants
import { AGGREGATION_TYPE } from "@/constants";

export const mockSalaryData = [
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
];

export const mockBillingCard = {
  id: 1,
  type: "Mastercard",
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};
