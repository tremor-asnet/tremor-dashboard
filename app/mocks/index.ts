export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
