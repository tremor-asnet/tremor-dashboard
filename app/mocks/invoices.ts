import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    date: "March, 21, 2021",
    code: "MS-123456",
    price: 180,
  },
  {
    date: "Apr, 22, 2022",
    code: "MS-123457",
    price: 180,
  },
  {
    date: "May, 23, 2023",
    code: "MS-999999",
    price: 180,
  },
  {
    date: "Jun, 24, 2024",
    code: "MS-888888",
    price: 180,
  },
  {
    date: "Jul, 25, 2025",
    code: "MS-476543",
    price: 180,
  },
];
