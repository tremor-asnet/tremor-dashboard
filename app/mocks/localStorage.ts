// Mock localStorage
export const mockLocalStorage = (() => {
  let store: Record<string, string> = {};

  return {
    getItem: jest.fn((key: string) => store[key] || null),
    setItem: jest.fn((key: string, value: string) => {
      store[key] = value.toString();
    }),
    removeItem: jest.fn((key: string) => {
      delete store[key];
    }),
    clear: jest.fn(() => {
      store = {};
    }),
  };
})();

Object.defineProperty(window, "localStorage", { value: mockLocalStorage });
