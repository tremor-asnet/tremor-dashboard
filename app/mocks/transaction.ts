// Types
import { Transaction } from "@/types";

export const MOCK_TRANSACTIONS: Transaction[] = [
  {
    id: 453119,
    createdAt: "2024-02-05T02:50:02.319095+00:00",
    amount: 2500,
    type: 1,
    status: 1,
    service: "Netflix",
  },
  {
    id: 124567,
    createdAt: "2024-02-05T02:50:49.176872+00:00",
    amount: 2000,
    type: 0,
    status: 1,
    service: "Apple",
  },
  {
    id: 5564477,
    createdAt: "2024-02-05T02:51:04.990179+00:00",
    amount: 750,
    type: 0,
    status: 1,
    service: "Stripe",
  },
  {
    id: 5443356,
    createdAt: "2024-01-05T02:51:28+00:00",
    amount: 1000,
    type: 0,
    status: 1,
    service: "HubSpot",
  },
  {
    id: 908978,
    createdAt: "2024-02-05T02:52:07.932255+00:00",
    amount: 2500,
    type: 0,
    status: 1,
    service: "Creative Tim",
  },
  {
    id: 445678,
    createdAt: "2024-02-05T02:52:39.920117+00:00",
    amount: 709,
    type: 0,
    status: 0,
    service: "Webflow",
  },
  {
    id: 664367,
    createdAt: "2024-02-05T02:52:56.283565+00:00",
    amount: 90,
    type: 0,
    status: 1,
    service: "Momo",
  },
];
