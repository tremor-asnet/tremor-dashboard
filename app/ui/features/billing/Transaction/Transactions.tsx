// Icons
import { FaRegCalendarAlt } from "react-icons/fa";

// Components
import { Text } from "@tremor/react";
import TransactionList from "../TransactionList/TransactionList";

// Types
import { OptionType, Transaction, User } from "@/types";

// Services
import { getUser } from "@/services";

// Components
import CreateTransactionContainer from "./CreateTransactionContainer";

interface TransactionsProps {
  newest: Transaction[];
  date: string;
}

const Transactions = async ({ newest, date }: TransactionsProps) => {
  const data: User[] = await getUser();

  const emailOptions: OptionType[] = data.map((user: User, index) => ({
    option: user.email,
    value: (index + 1).toString(),
  }));

  return (
    <div className="p-6 bg-white dark:bg-[#202940] rounded-lg shadow-md create-transaction-wrapper">
      <div className="flex justify-between">
        <div className="flex gap-4">
          <h6 className="font-semibold text-primary dark:text-white">
            Your Transaction&#39;s
          </h6>
          <CreateTransactionContainer options={emailOptions} />
        </div>
        <div className="flex items-center text-secondary">
          <FaRegCalendarAlt className="dark:text-lighter" />
          <Text className="dark:text-lighter ml-2">{date}</Text>
        </div>
      </div>
      <h6 className="transaction-list-heading dark:text-lighter">NEWEST</h6>
      <div className="overflow-y-auto max-h-[505px]">
        <TransactionList transactions={newest} />
      </div>
    </div>
  );
};

export default Transactions;
