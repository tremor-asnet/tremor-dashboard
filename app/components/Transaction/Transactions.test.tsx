// Libs
import { render } from "@testing-library/react";

// Components
import Transactions from "@/components/Transaction/Transactions";

// Mocks
import { MOCK_TRANSACTIONS } from "@/mocks/transaction";

describe("Transactions Component Testing", () => {
  const newest = MOCK_TRANSACTIONS.slice(0, 2);
  const yesterday = MOCK_TRANSACTIONS.slice(2);

  it("should match snapshot", () => {
    const { container } = render(
      <Transactions
        newest={newest}
        yesterday={yesterday}
        date="23 - 30 March 2020"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
