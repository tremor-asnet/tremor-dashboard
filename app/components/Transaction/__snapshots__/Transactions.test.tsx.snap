// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions Component Testing should match snapshot 1`] = `
<div>
  <div
    class="p-6 bg-white dark:bg-[#202940] rounded-lg shadow-md"
  >
    <div
      class="flex justify-between"
    >
      <h6
        class="font-semibold text-[#344767] dark:text-white"
      >
        Your Transactions
      </h6>
      <div
        class="flex items-center text-[#7b809a]"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
          />
        </svg>
        <p
          class="text-sm ml-2"
        >
          23 - 30 March 2020
        </p>
      </div>
    </div>
    <h6
      class="transaction-list-heading"
    >
      NEWEST
    </h6>
    <ul>
      <div
        class="flex p-4 justify-between items-center"
      >
        <div
          class="flex gap-4 items-center"
        >
          <button
            class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-[#F44335] border-[#F44335]"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
              />
            </svg>
          </button>
          <div>
            <h6
              class="text-sm font-semibold text-[#344767] dark:text-white mb-1"
            >
              Netflix
            </h6>
            <p
              class="text-xs text-[#7b809a]"
            >
              05 Feb 09:50 AM
            </p>
          </div>
        </div>
        <p
          class="text-sm font-semibold text-[#F44335]"
        >
          - $ 2500
        </p>
      </div>
      <div
        class="flex p-4 justify-between items-center"
      >
        <div
          class="flex gap-4 items-center"
        >
          <button
            class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-[#4CAF50] border-[#4CAF50]"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
              />
            </svg>
          </button>
          <div>
            <h6
              class="text-sm font-semibold text-[#344767] dark:text-white mb-1"
            >
              Apple
            </h6>
            <p
              class="text-xs text-[#7b809a]"
            >
              05 Feb 09:50 AM
            </p>
          </div>
        </div>
        <p
          class="text-sm font-semibold text-[#4CAF50]"
        >
          + $ 2000
        </p>
      </div>
    </ul>
    <h6
      class="transaction-list-heading"
    >
      YESTERDAY
    </h6>
    <ul>
      <div
        class="flex p-4 justify-between items-center"
      >
        <div
          class="flex gap-4 items-center"
        >
          <button
            class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-[#4CAF50] border-[#4CAF50]"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
              />
            </svg>
          </button>
          <div>
            <h6
              class="text-sm font-semibold text-[#344767] dark:text-white mb-1"
            >
              Stripe
            </h6>
            <p
              class="text-xs text-[#7b809a]"
            >
              05 Feb 09:51 AM
            </p>
          </div>
        </div>
        <p
          class="text-sm font-semibold text-[#4CAF50]"
        >
          + $ 750
        </p>
      </div>
      <div
        class="flex p-4 justify-between items-center"
      >
        <div
          class="flex gap-4 items-center"
        >
          <button
            class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-[#4CAF50] border-[#4CAF50]"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
              />
            </svg>
          </button>
          <div>
            <h6
              class="text-sm font-semibold text-[#344767] dark:text-white mb-1"
            >
              HubSpot
            </h6>
            <p
              class="text-xs text-[#7b809a]"
            >
              05 Jan 09:51 AM
            </p>
          </div>
        </div>
        <p
          class="text-sm font-semibold text-[#4CAF50]"
        >
          + $ 1000
        </p>
      </div>
      <div
        class="flex p-4 justify-between items-center"
      >
        <div
          class="flex gap-4 items-center"
        >
          <button
            class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-[#4CAF50] border-[#4CAF50]"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
              />
            </svg>
          </button>
          <div>
            <h6
              class="text-sm font-semibold text-[#344767] dark:text-white mb-1"
            >
              Creative Tim
            </h6>
            <p
              class="text-xs text-[#7b809a]"
            >
              05 Feb 09:52 AM
            </p>
          </div>
        </div>
        <p
          class="text-sm font-semibold text-[#4CAF50]"
        >
          + $ 2500
        </p>
      </div>
      <div
        class="flex p-4 justify-between items-center"
      >
        <div
          class="flex gap-4 items-center"
        >
          <button
            class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-[#5d6c85] border-[#5d6c85]"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.005 3.1a1 1 0 1 1 1.99 0l-.388 6.35a.61.61 0 0 1-1.214 0L7.005 3.1ZM7 12a1 1 0 1 1 2 0 1 1 0 0 1-2 0Z"
              />
            </svg>
          </button>
          <div>
            <h6
              class="text-sm font-semibold text-[#344767] dark:text-white mb-1"
            >
              Webflow
            </h6>
            <p
              class="text-xs text-[#7b809a]"
            >
              05 Feb 09:52 AM
            </p>
          </div>
        </div>
        <p
          class="text-sm font-semibold text-[#5d6c85]"
        >
          Pending
        </p>
      </div>
      <div
        class="flex p-4 justify-between items-center"
      >
        <div
          class="flex gap-4 items-center"
        >
          <button
            class="flex shrink-0 w-9 h-9 justify-center items-center border rounded-full text-[#4CAF50] border-[#4CAF50]"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6.293 13.293 1.414 1.414L12 10.414l4.293 4.293 1.414-1.414L12 7.586z"
              />
            </svg>
          </button>
          <div>
            <h6
              class="text-sm font-semibold text-[#344767] dark:text-white mb-1"
            >
              Momo
            </h6>
            <p
              class="text-xs text-[#7b809a]"
            >
              05 Feb 09:52 AM
            </p>
          </div>
        </div>
        <p
          class="text-sm font-semibold text-[#4CAF50]"
        >
          + $ 90
        </p>
      </div>
    </ul>
  </div>
</div>
`;
